addons/point_of_sale/models/pos_order.py

```diff
@@ -628,22 +628,9 @@ def _prepare_refund_data(self, refund_order_id):
     def create(self, values):
         if values.get('order_id') and not values.get('name'):
             # set name based on the sequence specified on the config
-            config_id = self.order_id.browse(values['order_id']).session_id.config_id.id
-            # HACK: sequence created in the same transaction as the config
-            # cf TODO master is pos.config create
-            # remove me saas-15
-            self.env.cr.execute("""
-                SELECT s.id
-                FROM ir_sequence s
-                JOIN pos_config c
-                  ON s.create_date=c.create_date
-                WHERE c.id = %s
-                  AND s.code = 'pos.order.line'
-                LIMIT 1
-                """, (config_id,))
-            sequence = self.env.cr.fetchone()
-            if sequence:
-                values['name'] = self.env['ir.sequence'].browse(sequence[0])._next()
+            config = self.order_id.browse(values['order_id']).session_id.config_id
```

Comment:
you don't need to change this line if you want to avoid your name to popup in this weird browse construction ;-)

Reply:
I needed the recordset, and not the id which was saved in the current code ^^.
But I'll take the opportunity to use a cleaner env['pos.order'] instead.